src/wui/load_or_save_game.cc

```diff
@@ -361,213 +333,180 @@ UI::Button* LoadOrSaveGame::delete_button() {
+			                                                    hour:minute. This is part of a list. */
+			   (boost::format(_("saved yesterday at %1%:%2%")) % savedate->tm_hour % minute).str();
+		} else {  // Older
+			gamedata.savedatestring =
+			   /** TRANSLATORS: Display date for choosing a savegame/replay. Placeholders are:
+			                                                    month day, year */
+			   (boost::format(_("%1% %2%, %3%")) % localize_month(savedate->tm_mon) %
+			    savedate->tm_mday % (1900 + savedate->tm_year))
+			      .str();
+			gamedata.savedonstring =
+			   /** TRANSLATORS: Display date for choosing a savegame/replay. Placeholders are:
+			                                                    month (short name) day (number), year
+			      (number). This is part of a list. */
+			   (boost::format(_("saved on %1% %2%, %3%")) % savedate->tm_mday %
+			    localize_month(savedate->tm_mon) % (1900 + savedate->tm_year))
+			      .str();
+		}
+	}
+}
 
-	FilenameSet gamefiles;
-	if (filetype_ == FileType::kReplay) {
-		gamefiles = g_fs->filter_directory(kReplayDir, [](const std::string& fn) {
-			return boost::algorithm::ends_with(fn, kReplayExtension);
-		});
-		// Update description column title for replays
-		table_.set_column_tooltip(2, show_filenames_ ? _("Filename: Map name (start of replay)") :
-		                                               _("Map name (start of replay)"));
-	} else {
-		gamefiles = g_fs->filter_directory(kSaveDir, [](const std::string& fn) {
-			return boost::algorithm::ends_with(fn, kSavegameExtension);
-		});
+void LoadOrSaveGame::add_general_information(SavegameData& gamedata,
+                                             const Widelands::GamePreloadPacket& gpdp) const {
+	gamedata.set_mapname(gpdp.get_mapname());
+	gamedata.set_gametime(gpdp.get_gametime());
+	gamedata.set_nrplayers(gpdp.get_number_of_players());
+	gamedata.version = gpdp.get_version();
+	gamedata.wincondition = gpdp.get_localized_win_condition();
+	gamedata.minimap_path = gpdp.get_minimap_path();
+}
+
+void LoadOrSaveGame::add_error_info(SavegameData& gamedata, std::string errormessage) const {
+	boost::replace_all(errormessage, "\n", "<br>");
+	gamedata.errormessage =
+	   ((boost::format("<p>%s</p><p>%s</p><p>%s</p>"))
+	    /** TRANSLATORS: Error message introduction for when an old savegame can't be loaded */
+	    % _("This file has the wrong format and can’t be loaded."
+	        " Maybe it was created with an older version of Widelands.")
+	    /** TRANSLATORS: This text is on a separate line with an error message below */
+	    % _("Error message:") % errormessage)
+	      .str();
+
+	gamedata.mapname = FileSystem::filename_without_ext(gamedata.filename.c_str());
+}
+
+void LoadOrSaveGame::add_sub_dir(const std::string& gamefilename) {
+	// Add subdirectory to the list
+	const char* fs_filename = FileSystem::fs_filename(gamefilename.c_str());
+	if (!strcmp(fs_filename, ".") || !strcmp(fs_filename, ".."))
```

Comment:
Please add the optional {} here, it's safer to always do that to prevent bugs.

We still have lots of there in the code base, because cleaning them all up at one time would be a huge undertaking. We don't want to add any new cases where they are missing though.

Reply:
done